src/core/diags/factory_diags.cpp

```diff
@@ -232,6 +233,7 @@ Error Diags::ProcessFrame(uint8_t aArgsLength, char *aArgs[])
     uint16_t size              = OT_RADIO_FRAME_MAX_SIZE;
     bool     securityProcessed = false;
     bool     csmaCaEnabled     = false;
+    bool     isHeaderUpdated   = false;
```

Comment:
I  think this should be updated to `true` when `-s` is supplied.

Reply:
Updated